tests/behat/features/fields.feature

```diff
@@ -228,3 +230,72 @@ Feature: Fields for Landing Page content type
     And I see field "Background Colour"
     And I should see an "select#edit-field-landing-page-bg-colour" element
     And I should see an "select#edit-field-landing-page-bg-colour.required" element
+
```

Comment:
@asarfara cc @anthony-malkoun Tide Landing Page only provides the content type and default components. It doesn't provide any extra features. It doesn't depend Tide API. This test is a responsibility of tide_automated_listing.

Reply:
Thats not true, it also allows connection to other components :

```
      $is_featured_news = ($bundle == 'featured_news' && $moduleHandler->moduleExists('tide_news'));
      $is_embedded_webform = ($bundle == 'embedded_webform' && $moduleHandler->moduleExists('tide_webform'));
      $is_automated_listing = ($bundle == 'automated_card_listing' && $moduleHandler->moduleExists('tide_automated_listing'));

      if ($is_featured_news || $is_embedded_webform || $is_automated_listing) {
```

Reply:
Tide Landing Page only provides tests for both Featured news and Embedded Webform in suggested mode when both Tide News and Tide Webform are installed. Tide News provides the Featured news paragraph but no test. Tide Webform doesn't provide the Embedded Webform component.

Featured News component conditionally depends on Tide News to reference the News content type, hence Tide Landing Page conditionally enables it.

Reply:
@sonnykt / @asarfara why do we have any dependency at all between `tide_landing_page` and `tide_automated_listing`? Landing pages are still valid without an automated listing and automated listings have use outside of landing pages? This begs the question of where should tests live that test listings on landing pages?

This has been traditionally done by using the composer `suggests` stanzas but I think this has caused more pain than it is worth. We're trying to introduce the concept of reference sites that would have configurations that would be integration tested. Can the tests live in this third place that is a site that has landing pages with automated listings on it?

How do Drupal modules handle this?

Reply:
hi @anthony-malkoun,

In my original design, tide_landing_page doesn't depend on tide_automated_listing and it has nothing to do with tide_automated_listing. In contrast, tide_automated_listing depends on tide_landing_page as it adds its component to the landing_page content type. Tide_automated_listing can work without tide_landing_page module, however, it doesn't make sense to do so as Landing Page is the only content type with the concept of "components". 

Tide Landing Page can be removed from the info.yml of tide_automated_listing without any problem. However, landing page content type (or a content type with a paragraph field) is still required for the test of tide_automated_listing so it could be a suggested package in  composer. 

The paragraph bundle (and all its functionalities) is provided by tide_automated_listing hence the module is responsible for testing. It doesn't make sense to provide features in a module and test them in another one.